sdk/appconfiguration/azure-appconfiguration-provider/samples/refresh_sample_feature_flags.py

```diff
@@ -0,0 +1,73 @@
+# ------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# -------------------------------------------------------------------------
+from azure.appconfiguration.provider import load, WatchKey
+from azure.appconfiguration import (
+    AzureAppConfigurationClient,
+    ConfigurationSetting,
+    FeatureFlagConfigurationSetting,
+)
+from sample_utilities import get_client_modifications
+import os
+import time
+
+kwargs = get_client_modifications()
+connection_string = os.environ.get("APPCONFIGURATION_CONNECTION_STRING")
+
+# Setting up a configuration setting with a known value
+client = AzureAppConfigurationClient.from_connection_string(connection_string)
+
+configuration_setting = ConfigurationSetting(key="message", value="Hello World!")
+feature_flag_setting = FeatureFlagConfigurationSetting("Beta", enabled=True)
+
+client.set_configuration_setting(configuration_setting=configuration_setting)
+client.set_configuration_setting(configuration_setting=feature_flag_setting)
+
+
+def my_callback_on_fail(error):
+    print("Refresh failed!")
+
+
+# Connecting to Azure App Configuration using connection string, and refreshing when the configuration setting message changes
+config = load(
+    connection_string=connection_string,
+    refresh_on=[WatchKey("message")],
+    refresh_on_feature_flags=True,
```

Comment:
I want to call out that refresh_on doesn't do anything here, unless refresh_on_feature_flags is false. Something like:

```suggestion
    refresh_on_feature_flags=True,
    refresh_on=[WatchKey("message")], # If refresh_on_feature_flags is false, message will continue to refresh.
```

Reply:
> I want to call out that refresh_on doesn't do anything here, unless refresh_on_feature_flags is false.

I'm not sure what you mean by this. You are able to refresh key/values and feature flags separately. You can turn off "refresh" so no refresh on, but still refresh feature flags. And the reverse for turning of refreshing feature flags and still refreshing key/values.

Reply:
Oh nevermind. I wrongly assumed "message" was also a feature flag.

Reply:
All feature flags are considered watch keys by default.